test/FastTests/Client/Subscriptions/LinqMethodsSupportForSubscriptionScript.cs

```diff
@@ -517,6 +517,315 @@ await session.StoreAsync(new SupportCall()
             }
         }
 
+        [Fact]
+        public async Task CanHandleDates_Today()
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(new SupportCall
+                    {
+                        Name = "Michael",
+                        Started = new DateTime(2000, 1, 1)
+                    });
+
+                    await session.StoreAsync(new SupportCall
+                    {
+                        Name = "Maxim",
+                        Started = DateTime.Now.AddYears(1)
+                    });
+
+                    await session.StoreAsync(new SupportCall
+                    {
+                        Name = "Aviv",
+                        Started = new DateTime(1986, 4, 19)
+                    });
+
+                    await session.SaveChangesAsync();
+                }
+
+                var options = new SubscriptionCreationOptions<SupportCall>
+                {
+                    Criteria = new SubscriptionCriteria<SupportCall>(
+                        call => call.Started > DateTime.Today)
```

Comment:
About actual usage: that's nice and readable and very convenient for basic usage.
But
I think that now that we are pushing much more javascript to our api, maybe we should expose it more, rather then hiding it. It's important both for "educational" and api consistency reasons. Moreover, we've seen in the past that duality between "basic" and "pro" APIs hides the true potential of tools from our users.